packages/core/src/fetchers/fetchAllEntities.ts

```diff
@@ -0,0 +1,96 @@
+import { ContentfulClientApi, Entry } from 'contentful';
+import { MinimalEntryCollection } from './gatherAutoFetchedReferentsFromIncludes';
+
+const MIN_FETCH_LIMIT = 1;
+
+const fetchEntities = async ({
+  entityType,
+  client,
+  query,
+}: {
+  entityType: 'Entry' | 'Asset';
+  client: ContentfulClientApi<undefined>;
+  query: { 'sys.id[in]': string[]; locale: string; limit: number; skip: number };
+}) => {
+  if (entityType === 'Asset') {
+    return client.getAssets({ ...query });
+  }
+
+  return client.withoutLinkResolution.getEntries({ ...query });
+};
+
+export const fetchAllEntities = async ({
+  client,
+  entityType,
+  ids,
+  locale,
+  skip = 0,
+  limit = 100,
+  responseItems = [],
+}: {
+  client: ContentfulClientApi<undefined>;
+  entityType: 'Entry' | 'Asset';
+  ids: string[];
+  locale: string;
+  skip?: number;
+  limit?: number;
+  responseItems?: Entry[];
+}) => {
+  try {
+    if (!ids.length || !client) {
+      return {
+        items: [],
+        ...(entityType === 'Entry' && { includes: [] }),
```

Comment:
Btw, `includes` seems to be an object with two keys `Entry` and `Asset` , not an array :) 
(i only noticed it on Friday, as I was assuming it was same shape as `.items[]`  🤷 ) 

![image](https://github.com/contentful/experience-builder/assets/113918657/1b0a1401-9479-42fd-b9e9-1dbd46e00619)



![image](https://github.com/contentful/experience-builder/assets/113918657/703356ec-3e57-40b9-8bcf-820dc6ac8351)